api/src/main/java/me/lokka30/treasury/api/common/event/EventBus.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.common.event;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public enum EventBus {
+    INSTANCE;
+
+    private Map<Class<?>, EventCaller> events = new ConcurrentHashMap<>();
```

Comment:
`EventCaller` is a generic but here is raw, is this a mistake?

Reply:
no this is not a mistake. I will probably make the `EventCaller` non generic.